src/System.Collections/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -247,9 +247,9 @@ private void CopyTo(KeyValuePair<TKey, TValue>[] array, int index)
                 throw new ArgumentNullException("array");
             }
 
-            if (index < 0 || index > array.Length)
+            if ((uint)index > (uint)array.Length)
```

Comment:
Why this change?  To me the behavior isn't as clear, and I doubt it makes any measurable perf difference.